src/lib.rs

```diff
@@ -907,7 +907,10 @@ impl Connection {
                     // Feed crypto data to the TLS state, if there's data
                     // available at the expected offset.
                     if space.crypto_stream.readable() {
-                        let buf = space.crypto_stream.recv_pop(std::usize::MAX)?;
+                        let mut buf =
+                            vec![0; space.crypto_stream.recv_size()];
+                        // TODO: what if fin is true?
```

Comment:
CRYPTO frames don't have a `fin` flag, so there's nothing we need to do (i.e. this comment can be removed).